js/app.js

```diff
@@ -144,12 +158,20 @@ function randomizeMap() {
 
 function randomizePlayerSetup() {
   const players = Array.from(State.playerList);
-  const factions = randomizeFactions();
+  const useBot = document.getElementById("use-bot").checked;
+  const botPlayers = (useBot ? randomizeBotPlayers() : []);
```

Comment:
I would recommend reworking the flow. Previously, the flow was:
1. Determine players
2. Randomize the factions for those players.

With your change, you're doing:
1. Determine human players
2. Determine bot players + bot factions
3. Randomize factions for players

When I actually think we can stick with the previous two-step flow (determine players, then randomize factions). I think this will be more flexible as we move into randomly selecting bot factions, as we can calculate their reach against player factions using the same logic. I also think it keeps the input/output of the logical steps more clean - frist, you select all players, then, you choose factions for those players. You would just have to add some logic to the randomizeFaction function to only choose Bot factions for bot players, and only choose non-Bot factions for human players.